src/main/java/build/buildfarm/worker/shard/ShardWorkerContext.java

```diff
@@ -311,7 +312,10 @@ private void matchInterruptible(MatchListener listener) throws IOException, Inte
       // transient backplane errors will propagate a null queueEntry
     }
     listener.onWaitEnd();
-    if (queueEntry == null || satisfiesRequirements(matchProvisions, queueEntry.getPlatform())) {
+
+    DequeueMatchSettings settings = new DequeueMatchSettings();
```

Comment:
Every time `workerContext.match()` is called for a new operation, this setting will be created again and in the end, it seems to be a couple of hardcoded booleans, which won't be changed during the lifetime of the worker. Maybe we can make this either a member of `ShardWorkerContext` or a singleton with a factory method so that we dont need to create it repeatedly?

Reply:
Yeah, we'll need bubble this up all the way to a proto config (ideally).  I'll do this next so we're not creating it in every match call